### PR TITLE
Added Individual Integration page 

### PR DESCRIPTION
**Pull request by sahilsuman933**
Signed-off-by: Sahil Suman <sahilsuman933@gmail.com>

**Description**
- Added Individual Integration page only for Docker
- Honeycomb Design is still left to do

This PR updates #2943

**Notes for Reviewers**

**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit 38161c626eb9ee50a9ceb3ad700b2fb7ab54235d at: https://62ed0bec39e463538dd58dbb--layer5.netlify.app

**Comment by leecalcote** [comment]
https://62ed0bec39e463538dd58dbb--layer5.netlify.app/service-mesh-management/meshery/integrations/docker

**Comment by l5io** [comment]
🚀 Preview for commit ed1af672219c7c39cef75b46fdd95701c03f6566 at: https://62ed28b04d1e7c6feec70291--layer5.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 5283227b1919f571fdff7e368a7834ee0f71e978 at: https://62ed30face17ee729d94a866--layer5.netlify.app

**Comment by leecalcote** [comment]
Feedback.


https://user-images.githubusercontent.com/7570704/183159427-fc4af22c-a8af-4881-8a14-c3d90e442472.mov



Example arrow expansion can be found on the /community/newcomers page.

**Comment by leecalcote** [comment]
Some great work here. Feedback:

1. “Looking for something similar?” --> “Explore other <name> integrations”
1. We’ll need to switch out the Landscape and Meshery logos for other visual graphics, but these are great placeholders.
1. We should have a “See All Integrations” button under the “Explore other <name> integrations” section.

File: gatsby-config.js [hunk 1]
```diff
@@ -391,6 +391,13 @@ module.exports = {
         name: "news",
       },
     },
+    {
+      resolve: "gatsby-source-filesystem",
+      options: {
+        path: `${__dirname}/src/collections/Integrations`,
```
Comment (Nikhil-Ladha) [inline gatsby-config.js:397]: No need to capitalize the directory name under collections, it can be `integrations`.

File: gatsby-node.js [hunk 2]
```diff
@@ -59,6 +59,7 @@ exports.createPages = async ({ actions, graphql, reporter }) => {
 
   // Create Pages
   const { createPage } = actions;
+  const integrationPostTemplate = path.resolve("src/templates/integrations.js");
```
Comment (Nikhil-Ladha) [inline gatsby-node.js:62]: Always add new variables to the end of older definitions. And, keep the definition style similar to others
i.e, 
```
       const integrationTemplate = path.resolve(
           "src/templates/integrations.js");
       );
```
P.s: This is not a "post" so just "integrationsTemplate" is fine

File: gatsby-node.js [hunk 3]
```diff
@@ -263,6 +271,19 @@ exports.createPages = async ({ actions, graphql, reporter }) => {
...
+  allIntegrations.forEach((integration) => {
+    createPage({
+      path: `/service-mesh-management/meshery/integrations/${integration.slug}`,
+      component: integrationPostTemplate,
+      context: {
+        slug: integration.slug,
+      },
+    });
+  });
+
```
Comment (Nikhil-Ladha) [inline gatsby-node.js:286]: Add new definitions to the end

File: src/sections/Meshery/Meshery-integrations/Individual-Integration/IntegrationPage.style.js [hunk 4]
```diff
@@ -0,0 +1,144 @@
+import styled from "styled-components";
```
Comment (Nikhil-Ladha) [inline src/sections/Meshery/Meshery-integrations/Individual-Integration/IntegrationPage.style.js:1]: File names need not have to be in `CamelCase` they can be smallcase

File: src/sections/Meshery/Meshery-integrations/Individual-Integration/IntegrationPage.style.js [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+
+export const Overview = styled.section`
+  display: flex;
+  padding: 0 2rem 5rem 2rem;
+  gap: 5rem;
+  justify-content: center;
+  h3 {
+    font-weight: 700;
```
Comment (Nikhil-Ladha) [inline src/sections/Meshery/Meshery-integrations/Individual-Integration/IntegrationPage.style.js:9]: No need for this, the global value of 600 is fine

File: src/sections/Meshery/Meshery-integrations/Individual-Integration/IntegrationPage.style.js [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+  @media (max-width: 700px) {
+    flex-direction: column-reverse;
+  } ;
+`;
+
+export const Feature = styled.div`
+  display: flex;
+  flex-direction: column;
+  h3 {
+    font-weight: 700;
```
Comment (Nikhil-Ladha) [inline src/sections/Meshery/Meshery-integrations/Individual-Integration/IntegrationPage.style.js:20]: Same as above

File: src/sections/Meshery/Meshery-integrations/Individual-Integration/IntegrationPage.style.js (hunk 4 above)
Comment (Nikhil-Ladha) [inline src/sections/Meshery/Meshery-integrations/Individual-Integration/IntegrationPage.style.js:1]: No need to export CSS for each section as individual divs.
Just export one `div` which should be the parent wrapper and use class for the remaining sections to apply CSS

File: src/sections/Meshery/Meshery-integrations/Individual-Integration/IntegrationPage.style.js [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+`;
+
+export const Content = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 0.5rem;
+  p {
+    line-height: 40px;
+    font-family: "Qanelas Soft", "Open Sans", sans-serif;
+    text-align: justify;
```
Comment (Nikhil-Ladha) [inline src/sections/Meshery/Meshery-integrations/Individual-Integration/IntegrationPage.style.js:40]: Please don't update global styles especially `font-family`.

File: src/sections/Meshery/Meshery-integrations/Individual-Integration/IntegrationPage.style.js [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+    margin: auto 0;
+  }
+`;
+
+export const Content = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 0.5rem;
+  p {
+    line-height: 40px;
```
Comment (Nikhil-Ladha) [inline src/sections/Meshery/Meshery-integrations/Individual-Integration/IntegrationPage.style.js:38]: 2rem would be enough

File: src/sections/Meshery/Meshery-integrations/Individual-Integration/IntegrationPage.style.js [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+  span {
+    display: flex;
+    gap: 2rem;
+    flex-wrap: wrap;
+    .btn {
+      font-weight: bold;
+      transition: all 0.5s ease;
+      background: #ebc017;
+      opacity: 0.8;
+      padding: 5px 2rem;
```
Comment (Nikhil-Ladha) [inline src/sections/Meshery/Meshery-integrations/Individual-Integration/IntegrationPage.style.js:51]: Make it `padding: 0.5rem 2rem`

File: src/sections/Meshery/Meshery-integrations/Individual-Integration/IntegrationPage.style.js [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+  h2 {
+    font-size: 2.5rem;
+    font-weight: 700;
+    text-align: center;
+  }
+  img {
+    margin-bottom: 0.5rem;
+  }
+  p {
+    font-family: "Qanelas Soft", "Open Sans", sans-serif;
```
Comment (Nikhil-Ladha) [inline src/sections/Meshery/Meshery-integrations/Individual-Integration/IntegrationPage.style.js:141]: Same as above

File: src/sections/Meshery/Meshery-integrations/Individual-Integration/index.js [hunk 11]
```diff
@@ -0,0 +1,143 @@
+import React from "react";
+import { Accordion,AccordionItem,AccordionTitle,AccordionBody,IconWrapper,OpenIcon,CloseIcon } from "../../../../reusecore/Accordion";
+import { IoIosArrowDown } from "@react-icons/all-files/io/IoIosArrowDown";
+import { IoIosArrowUp } from "@react-icons/all-files/io/IoIosArrowUp";
+import FaqSectionWrapper from "../../../General/Faq/faqSection.style";
+import { HeroSection, CTA ,Content,Feature,Overview } from "./IntegrationPage.style";
+import { Link } from "gatsby";
+import { Row, Col } from "../../../../reusecore/Layout";
+import Button from "../../../../reusecore/Button";
+import bookCover from "../../../../assets/images/learn/book-cover.png";
```
Comment (Nikhil-Ladha) [inline src/sections/Meshery/Meshery-integrations/Individual-Integration/index.js:10]: Use `StaticImage`

File: src/templates/integrations.js [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+  const [mounted, setMounted] = React.useState(false);
+  React.useEffect(() => {
+    setMounted(true);
+  }, []);
+
+  // prevents ssr flash for mismatched dark mode
+
+  if (!mounted) {
+    return <div style={{ visibility: "hidden" }}>Prevent Flash</div>;
+  }
```
Comment (Nikhil-Ladha) [inline src/templates/integrations.js:37]: This needs to be removed

File: src/collections/Integrations/docker.mdx [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+      question: "How can I request a new integration?",
+      ans: "Yes, as the service management plane, Meshery, provisions and manage the lifecycle of service meshes.",
+    },
+    {
+      question: "Does Meshery manage service meshes?",
+      ans: "Yes, as the service management plane, Meshery, provisions and manage the lifecycle of service meshes.",
+    },
+  ]
+  }
+}/>
```
Comment (Nikhil-Ladha) [inline src/collections/Integrations/docker.mdx:30]: If we would have wanted to render the data in this way itself, then why take the pain of using MDX files.
The body should only have the description of the integration and nothing more.
Rest all information should be present in the frontmatters.

**Review by Nikhil-Ladha — CHANGES_REQUESTED** [review]

File: src/sections/Meshery/Meshery-integrations/Individual-Integration/IntegrationPage.style.js (hunk 4 above)
Comment (sahilsuman933) [reply src/sections/Meshery/Meshery-integrations/Individual-Integration/IntegrationPage.style.js:1]: Do you mean it should be something like integrationpage.style.js, or are you talking about folder name, i.e. Individual-Integration?

File: src/collections/Integrations/docker.mdx (hunk 13 above)
Comment (sahilsuman933) [reply src/collections/Integrations/docker.mdx:30]: Can you give an example of that?

File: src/templates/integrations.js (hunk 12 above)
Comment (sahilsuman933) [reply src/templates/integrations.js:37]: From Line 23 to 37 or the whole file?

File: src/sections/Meshery/Meshery-integrations/Individual-Integration/IntegrationPage.style.js (hunk 4 above)
Comment (Nikhil-Ladha) [reply src/sections/Meshery/Meshery-integrations/Individual-Integration/IntegrationPage.style.js:1]: Yes, `integrationpage.style.js`

File: src/templates/integrations.js (hunk 12 above)
Comment (Nikhil-Ladha) [reply src/templates/integrations.js:37]: Only the selected lines, i.e, 28-37

File: src/sections/Meshery/Meshery-integrations/Individual-Integration/IntegrationPage.style.js (hunk 4 above)
Comment (sahilsuman933) [reply src/sections/Meshery/Meshery-integrations/Individual-Integration/IntegrationPage.style.js:1]: But most of the other style modules follow the camel case only. Look into different folders.

File: src/templates/integrations.js (hunk 12 above)
Comment (sahilsuman933) [reply src/templates/integrations.js:37]: Okay!

File: src/collections/Integrations/docker.mdx (hunk 13 above)
Comment (Nikhil-Ladha) [reply src/collections/Integrations/docker.mdx:30]: Ok, the format should be this:
```
---
title:
subtitle:
heroImg:
features: [..., ...., ...] (Can be written on each line)
---

<Import the button component />
Write the body here
```

File: src/sections/Meshery/Meshery-integrations/Individual-Integration/IntegrationPage.style.js (hunk 4 above)
Comment (Nikhil-Ladha) [reply src/sections/Meshery/Meshery-integrations/Individual-Integration/IntegrationPage.style.js:1]: Well actually, they follow `pascalCase`

File: src/collections/Integrations/docker.mdx (hunk 13 above)
Comment (sahilsuman933) [reply src/collections/Integrations/docker.mdx:30]: Okay!

File: src/sections/Meshery/Meshery-integrations/Individual-Integration/IntegrationPage.style.js (hunk 4 above)
Comment (Nikhil-Ladha) [reply src/sections/Meshery/Meshery-integrations/Individual-Integration/IntegrationPage.style.js:1]: So, `integrationPage.style.js` should be the one

Comment (sahilsuman933) [reply src/sections/Meshery/Meshery-integrations/Individual-Integration/IntegrationPage.style.js:1]: Okay!

**Comment by Nikhil-Ladha** [comment]
Also, I noticed that the complete honeycomb is not clickable and it is only the image that is, fix that.
And, in mobile view, the honeycomb size (width/height) can be reduced, the current one is too big for mobile view.

**Comment by sahilsuman933** [comment]
@Nikhil-Ladha, I will fix that.

File: src/templates/integrations.js [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+
+  return (
+    <ThemeProvider theme={theme === "dark" ? darktheme : lighttheme}>
+      <Layout>
+        <GlobalStyle />
+        <SEO title="Integrations" />
+        <Navigation
+          theme={data.mdx.frontmatter.title}
```
Comment (Nikhil-Ladha) [inline src/templates/integrations.js:45]: It should be `theme={theme}` and the title of the integration :)

Comment (sahilsuman933) [reply src/templates/integrations.js:45]: Yeah, sorry for that. I will fix that. :/

**Comment by sahilsuman933** [comment]
@leecalcote, don't worry about the broken link for now because when this PR is merged, I will fix all of them.

File: gatsby-node.js [hunk 15]
```diff
@@ -208,6 +211,11 @@ exports.createPages = async ({ actions, graphql, reporter }) => {
            }
          }
        }
+       allIntegrations: allMdx(filter: { fields: { collection: { eq: "integrations" } } }) {
+        nodes {
+          slug
+        }
+      }
```
Comment (Nikhil-Ladha) [inline gatsby-node.js:218]: Rebase the PR, and re-use the same query that is defined in the beginning i.e, `allPosts` and filter like any other collections like:
`node = > node.filter.collections === "integrations"`

Comment (sahilsuman933) [reply gatsby-node.js:218]: Okay!

**Comment by leecalcote** [comment]
@sahilsuman933 heads-up: merge conflict here.

**Comment by Nikhil-Ladha** [comment]
What happened here? 🤔

**Comment by sahilsuman933** [comment]
I didn't know how to rebase the local repository. So I tried the sync fork in my GitHub repository of layer5 and pulled the updates, and I lost all the changes I made on my local computer, and all the commits I made got deleted. Is there any way through which I can recover those changes? :sob: :sob: :sob:

**Comment by leecalcote** [comment]
@sahilsuman933 there might be… if you can share the history of commands executed, we can discern whether those commits at hidden somewhere or not.

**Comment by sahilsuman933** [comment]
@leecalcote Thanks to @Nikhil-Ladha, I was able to recover my commits.  :)

**Comment by leecalcote** [comment]
Oh, geez. 🙄 Well, la-di-da.

Of course, the inaugural MeshMate of the Year @Nikhil-Ladha swoops in and saves the day. 🦸‍♂️Just like that. Inadvertently and quietly flexing his 🤓 💪 all over the place, being kind to people, helping his fellow contributor... This kind of behavior 🤢 make me want to 🤮 . 

😁

**Comment by Nikhil-Ladha** [comment]
Haha 😄 
What can I say, it has become a habit now 😎

**Pull request merged**